src/tenant/admin.py

```diff
@@ -115,6 +127,79 @@ def get_queryset(self, request):
                     tenant.update_cached_fields()
         return qs
 
+    @admin.action(description="Send an email message to all owners for the selected tenant(s)")
+    def send_email_message(self, request, queryset):
+        """
+        Send an email message to all owners for selected tenant(s).
+
+        This action first displays an intermediate page which shows compose message form and
+        a list of all recipients.
+
+        Next, it send email message to all selected users and redirects back to the change list.
+        """
+        from siteconfig.models import SiteConfig
+
+        # get a list of selected tenant(s), excluding public schema
+        objects = self.model.objects.filter(
+            pk__in=[str(x) for x in request.POST.getlist(helpers.ACTION_CHECKBOX_NAME)]).exclude(
+                schema_name=get_public_schema_name())
+
+        # get a list of all owners (recipients) for selected tenant(s)
+        owners = []
+        for tenant in objects:
+            with tenant_context(tenant):
+                config = SiteConfig.get()
+                if config.deck_owner is not None:
+                    owners.append(config.deck_owner)
+        owners = set(owners)
+
+        if request.POST.get("post"):  # if admin pressed 'post' on intermediate page
+            form = SendEmailAdminForm(data=request.POST)
+            if form.is_valid():
+                cleaned_data = form.cleaned_data
+
+                recipient_list = [o.email for o in owners if o.email]
+                # run background task that will send email messages
+                send_email_message.apply_async(
+                    # subject, message and recipient_list (emails)
+                    args=[cleaned_data["subject"], cleaned_data["message"], recipient_list],
```

Comment:
is the recipient list bcc?  It should be bcc, and can send direct to the project's admin email (in setting.py)

Reply:
Please reply to my comment.  I'm not sure why this was marked as resolved.